test/testdata/lsp/hover.rb

```diff
@@ -146,10 +146,26 @@ def main
       # ^^^^^^ hover: The docs for BigFoo
 
   foo = BigFoo.new
-  #            ^^^ hover: sig { returns(BigFoo) }
+  #            ^^^ hover: sig { void }
```

Comment:
It's a little unfortunate to see the type here turn to `void`, but it is indeed the correct type for `initialize`.

Reply:
Not pictured in this `hover` assertion is that on the line below in the true hover result there will be a `# result type:` line that has the type being `BigFoo`. The full hover result will be

````
```ruby
# BigFoo#initialize
sig { void }
private def initialize; end

# result type:
BigFoo
```
````

So users will be able to see the type that the expression evaluates to independently of the way that the signature is defined.

Reply:
I think that arguably, this is a better experience for those new to Ruby, to drive home the fact that the relevant method to look at for a call to `new` might be `initialize`—Sorbet will only show the signature here when the class did not have a custom `def self.new` override, so in the case where that method *has* been defined, Sorbet will be able to call attention to it.